packages/esm-patient-chart-app/src/encounter-list/components/encounter-list.component.tsx

```diff
@@ -0,0 +1,296 @@
+import { Button, Link, OverflowMenu, OverflowMenuItem, DataTableSkeleton, Pagination } from '@carbon/react';
+import { Add } from '@carbon/react/icons';
+import { launchEncounterForm } from '../utils/helpers';
+import { deleteEncounter } from '../encounter-list.resource';
+import { useEncounterRows, useFormsJson } from '../hooks';
+
+import styles from './encounter-list.scss';
+import { type TableRow, type Encounter, type Mode } from '../types';
+
+export interface EncounterListColumn {
+  key: string;
+  header: string;
+  getValue: (encounter: Encounter) => string;
+  link?: any;
+}
+
+export interface EncounterListProps {
+  patientUuid: string;
+  encounterType: string;
+  columns: Array<any>;
+  headerTitle: string;
+  description: string;
+  formList?: Array<{
+    name?: string;
+    uuid: string;
+    excludedIntents?: Array<string>;
+    fixedIntent?: string;
+    isDefault?: boolean;
+  }>;
+  launchOptions: {
+    hideFormLauncher?: boolean;
+    displayText?: string;
+    workspaceWindowSize?: 'minimized' | 'maximized';
+  };
+  filter?: (encounter: Encounter) => boolean;
+  afterFormSaveAction?: () => void;
+  deathStatus?: boolean;
+  currentVisit: Visit;
+}
+
+export const EncounterList: React.FC<EncounterListProps> = ({
+  patientUuid,
+  encounterType,
+  columns,
+  headerTitle,
+  description,
+  formList,
+  filter,
+  launchOptions,
+  afterFormSaveAction,
+  currentVisit,
+  deathStatus,
+}) => {
+  const { t } = useTranslation();
+
+  const [currentPage, setCurrentPage] = useState(1);
+  const [pageSize, setPageSize] = useState(10);
+
+  const { formsJson, isLoading: isLoadingFormsJson, error: errorFormJson } = useFormsJson(formList[0].uuid);
+  const { encounters, total, isLoading, onFormSave, mutate } = useEncounterRows(
+    patientUuid,
+    encounterType,
+    filter,
+    afterFormSaveAction,
+    pageSize,
+    currentPage,
+  );
+
+  const { workspaceWindowSize, displayText, hideFormLauncher } = launchOptions;
+
+  const defaultActions = useMemo(
+    () => [
+      {
+        label: t('viewEncounter', 'View'),
+        form: {
+          name: formsJson?.name,
+        },
+        mode: 'view',
+        intent: '*',
+      },
+      {
+        label: t('editEncounter', 'Edit'),
+        form: {
+          name: formsJson?.name,
+        },
+        mode: 'edit',
+        intent: '*',
+      },
+      {
+        label: t('deleteEncounter', 'Delete'),
+        form: {
+          name: formsJson?.name,
+        },
+        mode: 'delete',
+        intent: '*',
+      },
+    ],
+    [formsJson, t],
+  );
+
+  const createLaunchFormAction = useCallback(
+    (encounter: Encounter, mode: Mode) => () => {
+      launchEncounterForm(formsJson, currentVisit, mode, onFormSave, encounter.uuid, null, patientUuid);
+    },
+    [formsJson, onFormSave, patientUuid, currentVisit],
+  );
+
+  const handleDeleteEncounter = useCallback(
+    (encounterUuid, encounterTypeName) => {
```

Comment:
Can we specify the types of these two as `string` (or the correct type, if I'm mistaken)?